### DISCUSSION TITLE
Can't edit any scripts with VS Code

### DISCUSSION THREAD

**Discussion by brettshepherd** [johnlindquist/kit#700] [Errors]
Brand new to Script Kit and was trying to edit one of the example scripts to learn a bit, but I can't seem to get vscode to open. I've tried running the "change editor" and "sync $path" commands, to no avail. 

The path for vscode is in my ~/.zshrc file. 
`export PATH="/Applications/Visual Studio Code.app/Contents/Resources/app/bin/:${PATH}" `

The ~/.kenv/.env file after running the above commands is the following:
```
KIT=/Users/brettshepherd/.kit
KENV=/Users/brettshepherd/.kenv
KIT_TEMPLATE=default
KIT_EDITOR=code
PATH=/Applications/Visual Studio Code.app/Contents/Resources/app/bin/:/Library/Frameworks/Mono.framework/Versions/Current/bin/:/Users/brettshepherd/google-cloud-sdk/bin:/opt/homebrew/bin:/Users/brettshepherd/.nvm/versions/node/v16.13.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/:/Users/brettshepherd/.kit/node/bin:/Users/brettshepherd/.kit/bin:/Users/brettshepherd/.kit/bin/code:/Users/brettshepherd/.kenv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

Here's the exact error from Script Kit when trying to open a script:
```
Command failed with exit code 127: code --goto /Users/brettshepherd/.kenv/kenvs/examples/scripts/resize-image.js:0:0 --folder-uri /Users/brettshepherd/.kenv
/bin/sh: code: command not found
Error: Command failed with exit code 127: code --goto /Users/brettshepherd/.kenv/kenvs/examples/scripts/resize-image.js:0:0 --folder-uri /Users/brettshepherd/.kenv
/bin/sh: code: command not found
    at makeError (file:///Users/brettshepherd/.kit/node_modules/@johnlindquist/globals/dist/index.js:15:247182)
    at execaSync$1 (file:///Users/brettshepherd/.kit/node_modules/@johnlindquist/globals/dist/index.js:15:259137)
    at execaCommandSync$1 (file:///Users/brettshepherd/.kit/node_modules/@johnlindquist/globals/dist/index.js:15:259633)
    at code (file:///Users/brettshepherd/.kit/platform/darwin.js:155:5)
    at global.edit (file:///Users/brettshepherd/.kit/platform/darwin.js:179:24)
    at async file:///Users/brettshepherd/.kit/cli/app-run.js
```

**Comment by johnlindquist** [comment]
Does running `code` work in your terminal?

In the .env, try changing the editor to full path:

```
KIT_EDITOR=/usr/local/bin/code
```

If that works, then we can try to debug the PATH lookup.

**Reply by brettshepherd** [reply]
The `code` command does work fine in terminal.

Using the `which` command in the terminal i get the following location:
```
 which code
/Applications/Visual Studio Code.app/Contents/Resources/app/bin//code
```

I tried setting that location as the KIT_EDITOR, but now instead of getting an error message, it just closes the Script Kit window and nothing happens.

Tried the following combinations just to see if anything would change: 
```
KIT_EDITOR=/Applications/Visual Studio Code.app/Contents/Resources/app/bin//code
KIT_EDITOR=/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code
KIT_EDITOR=/Applications/Visual Studio Code.app/Contents/Resources/app/bin
```

**Reply by johnlindquist** [reply]
## "Standard" setup

https://code.visualstudio.com/docs/setup/mac#_launching-from-the-command-line

Following the instructions should produce:

`which code` = `/usr/local/bin/code`

This path doesn't have an spaces. It looks like you configured the .zshrc by hand instead of using the method provided by vs code.

## Spaces in Paths for Custom Editors
I believe the root cause of using `/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code` is the spaces in `/Visual Studio Code/`. Unfortunately, the only fix for now would be patching kit until the next release:

Open: `/Users/johnlindquist/.kit/platform/darwin.js`

Replace line 236 with:

```
// notice the new quotes
let editCommand = `"${KIT_EDITOR}" ${file}`;
```

Then the spaces shouldn't matter in the tool path.

## My action items
1. Commit the patch above
2. Figure out why your `PATH` with `KIT_EDITOR=code` worked on my machine, but not yours

**Reply by brettshepherd** [reply]
Thanks, I ended up just removing the `/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code` from my .zshrc and following your link's instructions to properly install the vscode path at `/usr/local/bin/code` and it now works!

**Reply by johnlindquist** [reply]
Awesome, glad that's sorted. Let me know if there's anything else I can help with!
